### DISCUSSION TITLE
KStar 3-phase Inverter / Missing registers in kstar-hybrid.yaml

### DISCUSSION THREAD

**Discussion by mknausde** [StephanJoubert/home_assistant_solarman#473] [General]
Hi all, Hi @frsantos, 

I have a KSTAR E10KT 3-Phase Inverter. First of all great job @frsantos for providing the existing template - so far it is working like a charm. But I am missing S & T Phase registers, which after some research are needed, to calculate the correct total energy consumption value. You already mentioned in the template comment, that it seems to be calculated from different register values and you are right. After closely comparing inverter values with the Solarman web portal and the app against available registers, I'm pretty sure that the correct Total Energy Consumption value is calculated from: Consumption Active Power R, Consumption Active Power S and Consumption Active Power T. Right now, the Total Consumption Power value seems to mimic the R-Phase Load Power register. 

Now while you are reading the R-Phase registers, I would need to get S & T Phase registers as well. I guess I could add them by myself if you are short on time, but I am unable to get my hands on the "MODBUS RS485 Communication Protocol V2.5" you mentioned, to query the right registers. Can you, or anyone provide that parameter definition?

**Comment by frsantos** [comment]
Hi @mknausde 

Exactly. As I didn't have the 3-phase inverter, I didn't add the missing registers.

The PDF is on the original PR: https://github.com/StephanJoubert/home_assistant_solarman/pull/253#issue-1640613352. Adding the new phases should be pretty straightforward, but summing them into a total it is not as easy, I think, as discussed in the PR.

**Reply by mknausde** [reply]
Thanks for your fast feedback and hinting me to the document. I have added the additional registers for S- and T-Phase without problems. For now I've created a helper in Home Assistant to sum up the 3 sensors (R/S/T-Load power) and to provide it as a separate sensor.

**Comment by PHei66** [comment]
Hi @mknausde 

I'm owner of the 3-phase inverter since a few days and I'm also missing the S and T registers. Can you please show me how to add them in the kstar-hybrid.yaml?
And are there any other todos or do I only have to change the yaml-file?

Thank you for your support

**Reply by mknausde** [reply]
HI @PHei66 

I'm sorry... just stumbled over your request, totally forgot to give you feedback. 
Do you still need the kstar.yaml? 

In short, you just add the missing lines with the correct modbus registers for the missing phases in the yaml.
Then you need to combine them for Load Power and Grip Power. I did this with helpers. The combined values of R/S/T Load Power and R/S/T Grid Power give you the correct readings.


Regards
